gulpfile.js

```diff
@@ -0,0 +1,164 @@
+const fs = require('fs');
+const path = require('path');
+const {dest, src, series} = require('gulp');
+const replace = require('gulp-replace');
+const rename = require('gulp-rename');
+const clean = require('gulp-clean');
+const readlineSync = require('readline-sync');
+const {execSync} = require('child_process');
+
+const E2E_PROJECT_NAME = 'fabric8-test';
+const E2E_REPO = 'https://github.com/fabric8io/fabric8-test.git';
+
+const E2E_PROJECT = path.resolve(__dirname, '../', E2E_PROJECT_NAME);
+const TEST_DIR = path.resolve(E2E_PROJECT, 'ee_tests');
+
+const CONFIG_DIR = path.resolve(TEST_DIR, 'config');
+const CONFIG_NAME = 'local_osio.conf.sh';
+const CONFIG = path.resolve(CONFIG_DIR, CONFIG_NAME);
+const CONFIG_TEMPLATE_NAME = 'local_osio.conf.sh.template';
+const CONFIG_TEMPLATE = path.resolve(CONFIG_DIR, CONFIG_TEMPLATE_NAME);
+
+// TODO read protractor.config.ts to identify test keys instead of hard coding
+const testSuites = ['all', 'che', 'local', 'logintest', 'planner', 'smoketest'];
+const testUrls = {
+  'enter custom': null,
+  dev: 'http://localhost:3000',
+  'local prod': 'http://localhost:8080',
+  production: 'http://openshift.io',
+  'prod-preview': 'http://prod-preview.openshift.io',
+};
+
+// only beta and released levels are supported
+const featureLevels = ['beta', 'released'];
+
+function e2eClean() {
+  if (fs.existsSync(E2E_PROJECT)) {
+    console.log(`Deleting project ${E2E_PROJECT}`);
+    return src(E2E_PROJECT, {read: false}).pipe(clean({force: true}));
+  }
+  return Promise.resolve();
+}
+
+// fetch and update e2e tests
+async function e2eFetch() {
```

Comment:
Failed for me because I already had `fabric8-test` checked out and was on a different branch. It would be safer to run `git checkout master` before pulling changes.

Reply:
argh! a silly mistake. Thanks for catching.